src/script/scriptBuilder.zig

```diff
@@ -0,0 +1,79 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const Script = @import("lib.zig").Script;
+pub const engine = @import("engine.zig");
+const Opcode = @import("./opcodes/constant.zig").Opcode;
+const ScriptNum = Script.ScriptNum;
+const asBool = Script.asBool;
+const asInt = Script.asInt;
+const testing = std.testing;
+///ScriptBuilder is a library to generate easier scripts, useful for faster testing
+pub const ScriptBuilder = struct {
+    /// Number of opcodes in the script(this is not a necessary field and may be removed later)
+    opcodeCount: u8,
+
+    /// Dynamic array holding the opcodes
+    opcodes: []u8,
+
+    ///Memory Allocator
+    allocator: Allocator,
+
+    /// Initialize a new ScriptBuilder
+    ///
+    /// # Arguments
+    /// - `allocator`: Memory allocator for managing engine resources
+    ///
+    /// # Returns
+    /// - `!ScriptBuilder`: use with `try`
+    pub fn new(allocator: std.mem.Allocator) !ScriptBuilder {
+        return ScriptBuilder{
+            .opcodeCount = 0,
+            .opcodes = try allocator.alloc(u8, 0),
+            .allocator = allocator,
+        };
+    }
+
+    ///Push an OPcode to the ScriptBuilder
+    pub fn pushOpcode(self: *ScriptBuilder, op: Opcode) !*ScriptBuilder {
```

Comment:
rename `addOpcode`

Reply:
Also, every time you expend the scrip, you should make sure the script doesn't exceed it's max canonical size, which is `10000`, 10k bytes